src/modm/driver/display/st7586s_protocol.hpp

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2021, Tomasz Wasilczyk
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+#ifndef MODM_ST7586S_HPP
+#error "Do not include this file directly. Include st7586s.hpp instead."
+#endif
+
+#include <modm/math/utils/endianness.hpp>
+
+namespace modm::impl::st7586s
+{
+
+enum class Command : uint8_t {
+	Nop = 0x00,
+	Reset = 0x01,
+	SleepOn = 0x10,
+	SleepOff = 0x11,
+	PartialOn = 0x12,
+	PartialOff = 0x13,
+	InverseOff = 0x20,
+	InverseOn = 0x21,
+	AllPixelsOn = 0x22,
+	AllPixelsOff = 0x23,
+	DisplayOff = 0x28,
+	DisplayOn = 0x29,
+	SetColumn = 0x2A,
+	SetRow = 0x2B,
+	WriteDisplayData = 0x2C,
+	ReadDisplayData = 0x2E,
+	PartialDisplayArea = 0x30,
+	ScrollArea = 0x33,
+	DisplayControl = 0x36,
+	ScanDirection = 0x36,
+	StartLine = 0x37,
+	DisplayModeGray = 0x38,
+	DisplayModeMono = 0x39,
+	EnableDdram = 0x3A,
+	DisplayDuty = 0xB0,
+	FirstOutputCom = 0xB1,
+	FoscDivider = 0xB3,
+	PartialDisplay = 0xB4,
+	NLineInversion = 0xB5,
+	ReadModifyWriteOn = 0xB8,
+	ReadModifyWriteOff = 0xB9,
+	SetVop = 0xC0,
+	VopIncrease = 0xC1,
+	VopDecrease = 0xC2,
+	SetBias = 0xC3,
+	SetBooster = 0xC4,
+	SetVopOffset = 0xC7,
+	AnalogControl = 0xD0,
+	AutoReadControl = 0xD7,
+	OtpRdWrControl = 0xE0,
+	OtpControlOut = 0xE1,
+	OtpWrite = 0xE2,
+	OtpRead = 0xE3,
+	OtpSelectionControl = 0xE4,
+	OtpProgramming = 0xE5,
+	FrameRateGray = 0xF0,
+	FrameRateMono = 0xF1,
+	TempRange = 0xF2,
+	TempGradient = 0xF4,
+};
+
+namespace payload {
+
+struct SetColumnRow {
+	uint8_t startHi = 0;
+	uint8_t start;
+	uint8_t endHi = 0;
+	uint8_t end;
+
+	constexpr SetColumnRow(uint8_t start, uint8_t end) : start(start), end(end) {}
+};
+
+enum class DisplayControl : uint8_t {
+	ComInc = 0,
+	ComDec = (1 << 7),
+	SegInc = 0,
+	SegDec = (1 << 6),
+};
```

Comment:
You can use `#include <modm/math/utils/bit_constants.hpp>`
`enum class DisplayControl : uint8_t {`
`ComInc = 0,`
`ComDec = Bit7,`
`SegInc = 0,`
`SegDec = Bit6,`
`};`

Reply:
Will do.